### PR TITLE
feat(Wizard): support for full width and full height

### LINKED ISSUE #1761: Support for full width and full height wizard
https://github.com/patternfly/patternfly-next/pull/1721

`.pf-c-wizard` now supports `.pf-m-full-width` and `.pf-m-full-height`

### PR DESCRIPTION
**Pull request by boaz0**
**What**:

Add two new props to Wizard:
 - `isFullWidth` - set wizard to full width
 - `isFullHeight` - set wizard to full height

closes #1761 

//cc @mcoker

### CODE REVIEW

**Comment by mcoker** [comment]
@mcarrano yep, that covers it. Thanks. Sorry @boaz0 I meant to list this in the original description. We just want to add a `width` prop (like we do with modal), and one for `height`, too.

**Comment by boaz0** [comment]
@mcoker & @mcarrano no problem. I will update this PR.
Thanks.

**Comment by boaz0** [comment]
Sorry but I couldn't help myself and added height and width props as required.

File: packages/patternfly-4/react-core/src/components/Wizard/Wizard.tsx [hunk 1]
```diff
@@ -44,6 +44,14 @@ export type WizardStepFunctionType = (newStep: { id?: string | number; name: str
 export interface WizardProps {
   /** True to show the wizard */
   isOpen?: boolean;
+  /** True to set full height wizard */
+  isFullHeight?: boolean;
+  /** True to set full width wizard */
+  isFullWidth?: boolean;
+  /** Default width of the Modal. */
```
Comment (tlabaj) [inline packages/patternfly-4/react-core/src/components/Wizard/Wizard.tsx]: Can you update the comments here to say Wizard.

File: packages/patternfly-4/react-core/src/components/Wizard/Wizard.tsx [hunk 2]
```diff
@@ -330,7 +346,7 @@ class Wizard extends React.Component<WizardProps> {
         <FocusTrap focusTrapOptions={{ clickOutsideDeactivates: true }}>
           <Backdrop>
             <Bullseye>
-              <div {...rest} className={css(styles.wizard, className)} role="dialog" aria-modal="true" aria-labelledby={this.titleId} aria-describedby={description ? this.descriptionId : undefined}>
+              <div {...rest} style={{ width, height }} className={css(styles.wizard, isFullWidth && styles.modifiers.fullWidth, isFullHeight && styles.modifiers.fullHeight,className)} role="dialog" aria-modal="true" aria-labelledby={this.titleId} aria-describedby={description ? this.descriptionId : undefined}>
```
Comment (tlabaj) [inline packages/patternfly-4/react-core/src/components/Wizard/Wizard.tsx]: Can you add Logic here to always apply isFullWith when there is a width set and alway set the isFullHeight when the height is set.

File: packages/patternfly-4/react-core/src/components/Wizard/Wizard.md [hunk 3]
```diff
@@ -178,3 +178,57 @@ class ValidationWizard extends React.Component {
   }
 }
 ```
+
+### Full Height and Width Wizard
```
Comment (tlabaj) [inline packages/patternfly-4/react-core/src/components/Wizard/Wizard.md]: can you make this title sentence case. 

> Full height and width wizard

File: packages/patternfly-4/react-core/src/components/Wizard/Wizard.tsx (hunk 2 above)
Comment (boaz0) [reply packages/patternfly-4/react-core/src/components/Wizard/Wizard.tsx]: should I remove 'isFullWidth/Height` from props?

File: packages/patternfly-4/react-core/src/components/Wizard/Wizard.tsx (hunk 1 above)
Comment (boaz0) [reply packages/patternfly-4/react-core/src/components/Wizard/Wizard.tsx]: Thanks!

File: packages/patternfly-4/react-core/src/components/Wizard/Wizard.md (hunk 3 above)
Comment (boaz0) [reply packages/patternfly-4/react-core/src/components/Wizard/Wizard.md]: yes sure.

File: packages/patternfly-4/react-core/src/components/Wizard/Wizard.tsx (hunk 2 above)
Comment (tlabaj) [reply packages/patternfly-4/react-core/src/components/Wizard/Wizard.tsx]: No, they should remain in the props.  When width or height are applied.the take precedence over the isFullWidth/Height.  we need to apply isFullWidth/Height for the scenario when a size that is bigger the max size is applied, the isFullWidth/Height modifiers will force the max to be overridden.

File: packages/patternfly-4/react-core/src/components/Wizard/Wizard.tsx [hunk 4]
```diff
@@ -287,6 +303,8 @@ class Wizard extends React.Component<WizardProps> {
     const computedSteps: ComputedStep[] = this.initSteps(steps, activeStep);
     const firstStep = activeStep === flattenedSteps[0];
     const isValid = activeStep.enableNext !== undefined ? activeStep.enableNext : true;
+    const setFullWidth = isFullWidth || !!(width)
```
Comment (tlabaj) [inline packages/patternfly-4/react-core/src/components/Wizard/Wizard.tsx]: Shouldn't this be we want to set these booleans if the width/height exist :
```suggestion
    const setFullWidth = isFullWidth || width
    const setFullHeight = isFullHeight || height
```

**Comment by boaz0** [comment]
I see the same problem on netlify but on my local machine it works fine.

In regards to `null` as the default. This means that we are not going to inject `width` or `height` styling to `wizard` and they will be set by the default `width` and `height` values from patternfly core.

**Comment by mcarrano** [comment]
If that's the case @boaz0 , maybe it makes sense to change the description to "custom width/height of the wizard"? What you say makes sense, but I think the word "default" in the description confused me. The default size is really coming from Core.

**Comment by redallen** [comment]
Wizard docs are indeed broken on master, not sure which commit broke it. It happened recently and I'll try to get a fix out before lunch :)

**Comment by boaz0** [comment]
@mcarrano updated the docs to have "custom" and not "default"

**Comment by redallen** [comment]
So it looks like the fix is in `jschuler:patternfly-react/controlled-wizard`. It has to do with fixing imports/exports. I'll try to cherry pick some commits into here from there.

**Comment by redallen** [comment]
Previews fixed, enjoy! https://1786-pr-patternfly-react-patternfly.surge.sh/patternfly-4/components/wizard/

**Comment by boaz0** [comment]
Thank you all @mcoker @mcarrano and @redallen

**Pull request merged**